src/UI/class-admin-bar.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Parsely tweaks to WordPress admin bar
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\UI;
+
+use WP_Admin_Bar;
+use Parsely\Parsely;
+use Parsely\Utils;
+
+/**
+ * Render Parse.ly related buttons in the WordPress administrator top bar.
+ *
+ * @since 3.2.0
+ */
+final class Admin_Bar {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Register admin bar buttons.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		// Priority 201 to load after Core's admin bar secondary groups (200).
+		add_action( 'admin_bar_menu', array( $this, 'admin_bar_parsely_stats_button' ), 201 );
+	}
+
+	/**
+	 * Adds the `Parse.ly Stats` button on the admin bar when the current object is a post or a page.
+	 *
+	 * @param WP_Admin_Bar $admin_bar WP_Admin_Bar instance, passed by reference.
+	 *
+	 * @return void
+	 */
+	public function admin_bar_parsely_stats_button( WP_Admin_Bar $admin_bar ): void {
+		$current_object = $GLOBALS['wp_the_query']->get_queried_object();
+
+		if ( empty( $current_object ) ) {
+			return;
+		}
+
+		if ( ! empty( $current_object->post_type ) ) {
+			$post_type_object = get_post_type_object( $current_object->post_type );
+			if ( $post_type_object && current_user_can( 'edit_post', $current_object->ID ) && $post_type_object->show_in_admin_bar ) {
```

Comment:
`edit_post` is more restrictive than our "Post row actions" links:

https://github.com/Parsely/wp-parsely/blob/b667f6f707e215b267485658289f972274e2d54a/src/UI/class-row-actions.php#L54-L55

Should we also limit the post row actions to those with `edit_post` capabilities? As is, we show the links to "Contributors" (even for posts they haven't authored).

Reply:
I agree we should be consistent, and I would go down the road of being less restrictive (as we've been doing up until now). I've removed the `can_edit` post check and added the `has_trackable_status` that you suggest in the other comment. 

With this, we should have the same conditions on both admin bar and posts list.